### PR TITLE
SBE-3015: Tracing-Starter braucht Unterstützung für Resilience4J

### PR DESCRIPTION
**Pull request by denizengin87**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
     </scm>
 
     <properties>
-        <java.version>1.8</java.version>
+        <java.version>11</java.version>
```
Comment (mirkosertic) [inline pom.xml]: Do we really need to generate Java 11 bytecode here? The library must work with Java8+, but I don't think we need to force Java 11 as the target runtime environment here

**Comment by mirkosertic** [comment]
PRs can only be merged iff all automated checks are passing

File: pom.xml [hunk 2]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>2.0.3.RELEASE</version>
+        <version>2.1.5.RELEASE</version>
```
Comment (LeovR) [inline pom.xml]: 2.1.6.RELEASE is the latest version

File: pom.xml [hunk 3]
```diff
@@ -71,22 +77,44 @@
...
 
+        <dependency>
+            <groupId>io.github.resilience4j</groupId>
+            <artifactId>resilience4j-spring-boot2</artifactId>
+            <version>0.16.0</version>
+        </dependency>
+
+        <dependency>
+            <groupId>io.github.resilience4j</groupId>
+            <artifactId>resilience4j-timelimiter</artifactId>
```
Comment (LeovR) [inline pom.xml]: Do we need this dependency?

File: pom.xml [hunk 4]
```diff
@@ -71,22 +77,44 @@
...
             <artifactId>hystrix-core</artifactId>
-            <version>1.5.12</version>
+            <version>1.5.18</version>
             <optional>true</optional>
         </dependency>
 
+        <dependency>
+            <groupId>io.github.resilience4j</groupId>
+            <artifactId>resilience4j-spring-boot2</artifactId>
+            <version>0.16.0</version>
```
Comment (LeovR) [inline pom.xml]: This should be optional

File: src/main/java/de/thalia/boot/tracing/resilience4j/CircuitBreakerSpanAspect.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+@Aspect
+@Slf4j
+@RequiredArgsConstructor
+public class CircuitBreakerSpanAspect implements Ordered {
+
+    private final Tracer tracer;
+
+    @Override
+    public int getOrder() {
+        return Integer.MIN_VALUE;
```
Comment (LeovR) [inline src/main/java/de/thalia/boot/tracing/resilience4j/CircuitBreakerSpanAspect.java]: Ordered.HIGHEST_PRECEDENCE is the semantically better version

**Pull request merged**